core/src/main/scala/org/http4s/internal/parsing/Rfc5234.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.internal.parsing
+
+import cats.parse.{Parser => P, Parser1}
+
+/** Common rules defined in RFC5234 and imported into various HTTP
+  * specs.
+  *
+  * @see https://tools.ietf.org/html/rfc5234#appendix-B.1
+  */
+private[http4s] object Rfc5234 {
+  val alpha: P[Char] =
+    P.charIn(0x41.toChar to 0x5a.toChar).orElse1(P.charIn(0x61.toChar to 0x7a.toChar))
```

Comment:
I see some bitset magic.  Do these get optimized when stated like this, or should we express them with a predicate?

Reply:
Yes. These are merged into a single BitSet.